cumulusci/tasks/bulkdata/utils.py

```diff
@@ -101,6 +106,7 @@ def _wait_for_job(self, job_id):
         if result == "Failed":
             for state_message in messages:
                 self.logger.error(f"Batch failure message: {state_message}")
+            raise BulkDataException("Job Error", messages)
```

Comment:
I still feel that the decision to respond to the job status logically belongs to the Task, not here in the mixin. When I call `_wait_for_job()`, I'd expect an exception to be thrown if an HTTP error ensues, but otherwise to be returned the status, whatever it is.

That is, at this layer, exception => "cannot determine state of job at the API level". At the Task layer, exception => "the operation did not succeed".

Reply:
I've implemented a compromise in the code.

If the client is written naively (as DeleteData was) then it will throw an exception. If the caller wants to manage job state themselves, they can do that easily.